src/couch_file.erl

```diff
@@ -223,19 +223,33 @@ delete(RootDir, Filepath) ->
 
 
 delete(RootDir, Filepath, Async) ->
-    DelFile = filename:join([RootDir,".delete", ?b2l(couch_uuids:random())]),
+    DelFile = deleted_filename(RootDir, Filepath),
+    RenameOnDelete = config:get_boolean("couchdb", "rename_on_delete", false),
     case file:rename(Filepath, DelFile) of
-    ok ->
-        if (Async) ->
+        ok when RenameOnDelete ->
+            ok;
+        ok when Async ->
             spawn(file, delete, [DelFile]),
             ok;
-        true ->
-            file:delete(DelFile)
-        end;
-    Error ->
-        Error
+        ok ->
```

Comment:
diff would have been clearer if you hadn't mixed whitespace changes with functional changes.